File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        std::unordered_map<int,int> count;
        std::vector<int> res;
        std::vector<int> freq(nums.size()+1,0);
        for (int i:nums){
            count[i]=count[i]+1;
            freq[i]++;
        }
        for(int i=0;i<freq.size();i++){
            if(freq[i]>1){
                res.push_back(i);
            }
        }
        return res;
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        count={}
        res=[]
        freq=[0 for i in range(len(nums)+1)]
        for i in nums:
            count[i]=count.get(i,0)+1
            freq[i]+=1
        for i in range(len(freq)):
            if freq[i]>1:
                res.append(i)
        return res
        